### PR TITLE
Feature/#16 GitHub pull requests

### LINKED ISSUE #16: Create Smart Contract for GitHub pull requests

### PR DESCRIPTION
**Pull request by markspanbroek**
Closes #16 

Added Smart Contract for registering GitHub pull requests. The contract requests merged status and creator from the GitHub API using Oraclize. Unit tests and integration test are provided.

This pull request includes an update to the latest version of TruffleDevEnv.

### CODE REVIEW

File: src/app-contracts/contracts/PullRequests.sol [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    mapping (uint => string) pullRequestIdToCreator;
+    mapping (uint => bool) pullRequestIdToMergedState;
+    mapping (bytes32 => UserQuery) userQueries;
+    mapping (bytes32 => MergedQuery) mergedQueries;
+
+    function PullRequests(string repo_) {
+        repo = repo_;
+    }
+
+    function creators(uint pullRequestId) constant returns (string creator) {
```
Comment (svdo) [inline src/app-contracts/contracts/PullRequests.sol]: function `creators` (plural) returns string `creator` (singular) ?

File: src/app-contracts/test/test-networks.js [hunk 2]
```diff
@@ -4,8 +4,16 @@ const testNetworks = {
   42: 'kovan'
 }
 
-function isTestNetwork (web3) {
-  return !!testNetworks[web3.version.network]
+async function isTestNetwork (web3) {
```
[the remark this answers is not in the document]
Comment (markspanbroek) [reply src/app-contracts/test/test-networks.js]: That is technically speaking true, but I chose to mark it as async anyway :) The code returns a promise, therefore callers should be aware that it's an async function. Now they can see that by looking at the function signature.

File: src/app-contracts/contracts/PullRequests.sol (hunk 1 above)
Comment (markspanbroek) [reply src/app-contracts/contracts/PullRequests.sol]: Yes, I tried to mimic the auto-generated getter function for mappings, which follow this pattern. But now that you mention it, it is confusing. I will change that.

**Comment by markspanbroek** [comment]
Regarding validation failures: I always like to be explicit about them, therefore *throw*. Also to ensure that there are no side-effects.

**Comment by markspanbroek** [comment]
@svdo I renamed the function `creators` to `creator`. I also changed `users` to `user` in the Users contract.

Additionally, I updated to the latest version of TruffleDevEnv to support the latest yarn and node. You might need to update your yarn and node as well.

**Pull request merged**